src/engine/Service.cpp

```diff
@@ -200,6 +171,57 @@ ResultTable Service::computeResult() {
   return {std::move(idTable), resultSortedOn(), std::move(localVocab)};
 }
 
+std::optional<std::string> Service::getSiblingValuesClause() const {
+  if (siblingTree_ == nullptr) {
+    return std::nullopt;
+  }
+
+  auto siblingResult = siblingTree_->getResult();
+  if (siblingResult->size() >
+      RuntimeParameters().get<"service-max-value-rows">()) {
+    return std::nullopt;
+  }
+
+  std::vector<ColumnIndex> commonColumnIndices;
+  auto siblingVariables = siblingTree_->getVariableColumns();
+  std::string vars = "";
+  for (const auto& localVar : parsedServiceClause_.visibleVariables_) {
+    auto it = siblingVariables.find(localVar);
+    if (it == siblingVariables.end()) {
+      continue;
+    }
+    vars += it->first.name() + " ";
+    commonColumnIndices.push_back(it->second.columnIndex_);
+  }
+  vars.pop_back();
+  if (commonColumnIndices.size() > 1) {
+    vars = "(" + vars + ")";
+  }
+
+  std::string valuesClause = " { ";
+  for (size_t rowIndex = 0; rowIndex < siblingResult->size(); ++rowIndex) {
+    std::string row;
+    for (size_t i = 0; i < commonColumnIndices.size(); ++i) {
+      const auto& optionalString = ExportQueryExecutionTrees::idToStringAndType(
+          siblingTree_->getRootOperation()->getIndex(),
+          siblingResult->idTable()(rowIndex, commonColumnIndices[i]),
+          siblingResult->localVocab());
```

Comment:
Somehow the localVocab accessed here with `siblingResult->localVocab()` doesn't contain useful data.
As a result no values are written into the valuesClause.

Reply:
I'll have a look at this. I however don't suspect that the problem is the local vocab.

Reply:
Can you send me a reproducer (Dataset + Query) where something doesn't work as expected?
I just tried a simple example with one variable where this worked.

Reply:
Ok sry, here is a reproducable example query, i use it with Olympics Dataset but it doesn't use the local dataset anyway:

```
PREFIX schema: <http://schema.org/>
PREFIX imdb: <https://www.imdb.com/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX p: <http://www.wikidata.org/prop/>
SELECT ?imdb_id ?imdb_votes ?imdb_rating WHERE {
VALUES ?imdb_id { "tt0477348" "tt0118715" "tt0116282" } .
  SERVICE <https://qlever.cs.uni-freiburg.de/api/imdb> {
	?movie_imdb imdb:id ?imdb_id .
	?movie_imdb imdb:type "movie" .
    ?movie_imdb imdb:numVotes ?imdb_votes .
    ?movie_imdb imdb:averageRating ?imdb_rating .
  }
}
ORDER BY DESC(?imdb_votes)
```

Expected result: 3 rows with the given `imdb_id` values.
Actual result: 0 rows, Debug log shows that the sent Service-Query contains a Values Clause with `imdb_id` as key but no values.

Reply:
I figured it out....
Your code is absolutely working as it should, the problem was the implementation of the `VALUES` clause.
Basically you could neither evaluate it twice, nor read it from the cache during the same query. 
I have pushed a quick fix to your branch (just do a git pull in your local copy to get it) and everything should work as expected.